lib/friends/location.rb

```diff
@@ -23,21 +25,65 @@ def self.deserialization_expectation
     end
 
     # @param name [String] the name of the location
-    def initialize(name:)
+    def initialize(name:, nickname_str: nil)
       @name = name
+      @nicknames = nickname_str&.split(" #{NICKNAME_REFIX}") || []
     end
 
     attr_accessor :name
+    attr_reader :nicknames
 
     # @return [String] the file serialization text for the location
     def serialize
-      "#{SERIALIZATION_PREFIX}#{@name}"
+      Paint.unpaint("#{SERIALIZATION_PREFIX}#{self}")
```

Comment:
Ooh this is clever! Would you mind changing `self` to `to_s` to make it a little clearer what's going on?

Reply:
Yes, it would be a bit clearer, but robocop doesn't like that. Besides, `#{self}` is also used in the file `friend.rb` for the same purpose.

Reply:
Ah I didn't realize RuboCop complained and that we used this elsewhere. This is fine to stay the way you have it then!